frontend/js/src/listens/ListenCard.tsx

```diff
@@ -117,9 +117,12 @@ export default class ListenCard extends React.Component<
     oldProps: ListenCardProps,
     oldState: ListenCardState
   ) {
-    const { listen: oldListen } = oldState;
-    const { customThumbnail } = this.props;
-    const { listen } = this.state;
+    const { listen: oldListen } = oldProps;
+    const { customThumbnail, listen } = this.props;
+
+    if (Boolean(listen) && !isEqual(listen, oldListen)) {
+      this.setState({ listen });
+    }
```

Comment:
Why do we need to save the listen in the state?
If it's to access it below in the add to queue functions, you can use the props directly

Reply:
The `Link with Musicbrainz` feature updates listen in the state with the new linked track data, so we cannot directly use props everywhere. https://github.com/anshg1214/listenbrainz-server/blob/listening-queues/frontend/js/src/listens/ListenCard.tsx#L413-L425

Reply:
The `Link with Musicbrainz` feature updates the listen state on successfully linking. So we cannot directly use props everywhere. https://github.com/anshg1214/listenbrainz-server/blob/listening-queues/frontend/js/src/listens/ListenCard.tsx#L413-L425